apps/src/applab/designElements/photochooser.jsx

```diff
@@ -0,0 +1,226 @@
+          handleBorderWidthChange={this.props.handleChange.bind(
+            this,
+            'borderWidth'
+          )}
+          handleBorderColorChange={this.props.handleChange.bind(
+            this,
+            'borderColor'
+          )}
+          handleBorderRadiusChange={this.props.handleChange.bind(
+            this,
+            'borderRadius'
+          )}
+        />
+        <BooleanPropertyRow
+          desc={'hidden'}
+          initialValue={$(element).hasClass('design-mode-hidden')}
+          handleChange={this.props.handleChange.bind(this, 'hidden')}
+        />
+        <ZOrderRow
+          element={this.props.element}
+          onDepthChange={this.props.onDepthChange}
+        />
+      </div>
+    );
+  }
+}
+
+class PhotoChooserEvents extends React.Component {
+  static propTypes = {
+    element: PropTypes.instanceOf(HTMLElement).isRequired,
+    handleChange: PropTypes.func.isRequired,
+    onInsertEvent: PropTypes.func.isRequired
+  };
+
+  getPhotoselectEventCode() {
+    const id = elementUtils.getId(this.props.element);
+    const code =
+      'onEvent("' +
+      id +
+      '", "change", function() {\n' +
+      '  console.log("' +
+      id +
+      ' photo selected!");\n' +
+      '  console.log(getImageURL("' +
+      id +
+      '"));\n' +
+      '});\n';
+    return code;
+  }
+
+  insertPhotoselect = () =>
+    this.props.onInsertEvent(this.getPhotoselectEventCode());
+
+  render() {
+    const element = this.props.element;
+    const clickName = 'Change';
+    const clickDesc = 'Triggered when a photo is selected.';
+
+    return (
+      <div id="eventRowContainer">
+        <PropertyRow
+          desc={'id'}
+          initialValue={elementUtils.getId(element)}
+          handleChange={this.props.handleChange.bind(this, 'id')}
+          isIdRow={true}
+        />
+        <EventHeaderRow />
+        <EventRow
+          name={clickName}
+          desc={clickDesc}
+          handleInsert={this.insertPhotoselect}
+        />
+      </div>
+    );
+  }
+}
+
+export default {
+  PropertyTab: PhotoChooserProperties,
+  EventTab: PhotoChooserEvents,
+  themeValues: themeValues.photoSelect,
+
+  create: function() {
+    const element = document.createElement('label');
+    element.setAttribute('class', 'img-upload');
+    element.setAttribute('data-canonical-image-url', 'icon://fa-camera');
+    element.style.margin = '0px';
+    element.style.lineHeight = '1';
+    element.style.overflow = 'hidden';
+    element.style.wordWrap = 'break-word';
+    element.style.borderStyle = 'solid';
+
+    element.style.textRendering = 'optimizeSpeed';
+    elementLibrary.setAllPropertiesToCurrentTheme(
+      element,
+      designMode.activeScreen()
+    );
+    element.style.backgroundSize = 'contain';
+    element.style.backgroundRepeat = 'no-repeat';
+    element.style.backgroundPosition = '50% 50%';
+    element.style.backgroundImage =
+      'url(' +
+      assetPrefix.renderIconToString(
```

Comment:
This gave me a huge block of text in the HTML. I think you should be able to get the same functionality without that huge block of text by using a different method to produce the background image. The best way that I know of would be to use FontAwesome https://github.com/code-dot-org/code-dot-org/blob/aec8766b9d282d8399c61a65063bf03745710417/apps/src/templates/PendingButton.jsx#L34